packages/aio-metrics-client/src/index.js

```diff
@@ -14,49 +15,94 @@ function setMetricsURL (metricsPostURL) {
   metricsURL = metricsPostURL
 }
 
+/**
+ * Process batched metrics
+ */
 async function processBatchCounter() {
   batchTimerSet = false
   if (metricsURL) {
-    Object.keys(batchedMetrics).forEach(async (metricName) => {
-      console.log('Sending Batch metrics for ', metricName, batchedMetrics[metricName])
-      // ex. Sending Batch metrics for  request_count { '14257-chimera-stage': 2 }
-      const postBody = JSON.stringify({metric: metricName, data: batchedMetrics[metricName]})
-      delete batchedMetrics[metricName]
+    let processPromises = []
+
+    // Bulk processing
+    processPromises = processPromises.concat(Object.keys(batchedMetricsBulk).map((metricName) => {
+      console.log('Sending bulk batch metrics for ', metricName, batchedMetricsBulk[metricName])
+      // ex. Sending bulk batch metrics for  request_count { '14257-chimera-stage': 2 }
+      const postBody = JSON.stringify({metric: metricName, data: batchedMetricsBulk[metricName]})
+      delete batchedMetricsBulk[metricName]
       const reqData = {
         method: 'POST',
         headers: {
           'Content-Type': 'application/json'
         },
         body: postBody
       }
-      await fetch(metricsURL, reqData)
-    })
+      return fetch(metricsURL, reqData)
+    }))
+
+    // Individual processing
+    processPromises = processPromises.concat(batchedMetrics.map((metric) => {
+      console.log('Sending batch metrics for ', metric.metric, metric)
+      // ex. Sending batch metrics for  request_count { '14257-chimera-stage': 2 }
+      const postBody = JSON.stringify(metric)
+      const reqData = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: postBody
+      }
+      return fetch(metricsURL, reqData)
+    }))
+
+    await Promise.all(processPromises)
```

Comment:
Perhaps use Promise.allSettled here since these promises are all independent? If not a rejection would discard all other promises that are to be run after the rejection

Reply:
Yup good thinking, change just pushed